File: mpp-proiect-csharp-culbec/Server/StartServer.cs
// See https://aka.ms/new-console-template for more information

using log4net;
using Network.NetworkUtils;
using Persistence.Repository;
using Server.Service;

namespace Server;

internal static class StartServer
{
    private static readonly ILog Log = LogManager.GetLogger(typeof(StartServer));
    private const int DefaultPort = 8888;
    private const string DefaultHost = "127.0.0.1";

    public static void Main(string[] args)
    {
        Log.Info("Starting the server...");

        Log.Info("Loading the server configuration...");
        var properties = ServerProperties.GetProperties();

        if (!properties.ContainsKey("ConnectionString"))
        {
            Log.Error("Couldn't load the connection string!");
            return;
        }

        var port = DefaultPort;
        var hostname = DefaultHost;
        if (!properties.TryGetValue("Port", out var property))
        {
            Log.WarnFormat("Port couldn't be loaded. Using the default port: {0}", DefaultPort);
        }
        else
        {
            var result = int.TryParse(property, out port);
            if (!result)
            {
                Log.WarnFormat("Port property not a number. Using the default port: {0}", DefaultPort);
                port = DefaultPort;
            }
        }

        if (!properties.ContainsKey("Hostname"))
        {
            Log.WarnFormat("Hostname couldn't be loaded. Using the default hostname: {0}", DefaultHost);
        }
        else
        {
            var result = properties.TryGetValue("Hostname", out hostname);
            if (!result)
            {
                Log.WarnFormat("Hostname couldn't be loaded. Using the default hostname: {0}", DefaultHost);
                hostname = DefaultHost;
            }
        }

        Log.Info("Initializing the repositories...");
        var repositoryParticipant = new RepositoryParticipantDb(properties);
        //var repositoryUser = new RepositoryUserDb(properties);
        var repositoryUser = new RepositoryUserEfCore(properties);
        var repositoryRace = new RepositoryRaceDb(properties);

        Log.Info("Initializing the service...");
        var serviceParticipant = new ServiceParticipantImpl(repositoryParticipant);
        var serviceUser = new ServiceUserImpl(repositoryUser);
        var serviceRace = new ServiceRaceImpl(repositoryRace);
        var service = new ServiceImpl(serviceParticipant, serviceUser, serviceRace);

        Log.InfoFormat("Starting the server...");
        var server = new ProtobufServer(hostname!, port, service);
        try
        {
            server.Start();
        }
        catch (Exception e)
        {
            Log.ErrorFormat("Error when starting the server: {0}", e.Message);
        }
    }
}

File: mpp-proiect-java-culbec/MotorcycleContest/Server/src/main/java/StartServer.java
import CommonUtils.IService;
import ConnectionUtils.DBUtils;
import NetworkUtils.AbstractServer;
import NetworkUtils.ProtobufServer;
import NetworkUtils.ServerException;
import Repository.*;
import Service.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.IOException;
import java.util.Properties;

public class StartServer {
    private static final Logger logger = LogManager.getLogger(StartServer.class);
    private static final int DEFAULT_PORT = 8888;
    private static final String DEFAULT_HOST = "localhost";

    public static void main(String[] args) throws RepositoryException {
        logger.traceEntry("Starting the server...");

        logger.info("Loading the server configuration...");
        Properties properties = new Properties();

        try {
            properties.load(StartServer.class.getResourceAsStream("/server.properties"));

            logger.info("Properties loaded successfully.");
            //properties.list(logger);
        } catch (Exception e) {
            logger.error("Error loading the server configuration: {}", e.getMessage());
            return;
        }

        logger.info("Initializing a DBUtils object...");
        try (DBUtils dbUtils = new DBUtils(properties)) {
            logger.info("Initializing the repositories...");
            UserRepository userRepository = new UserDBRepository(dbUtils);
            ParticipantRepository participantRepository = new ParticipantDBRepository(dbUtils);
            RaceRepository raceRepository = new RaceDBRepository(dbUtils);

            logger.info("Initializing the server...");

            ServiceUser serviceUser = new ServiceUserImpl(userRepository);
            ServiceParticipant serviceParticipant = new ServiceParticipantImpl(participantRepository);
            ServiceRace serviceRace = new ServiceRaceImpl(raceRepository);
            IService service = new ServiceImpl(serviceUser, serviceParticipant, serviceRace);

            logger.info("Retrieving the host and port of the server from the configuration...");
            String hostname = properties.getProperty("server.host", DEFAULT_HOST);
            int port = DEFAULT_PORT;

            try {
                port = Integer.parseInt(properties.getProperty("server.port"));
            } catch (NumberFormatException e) {
                logger.error("Invalid port number. Using the default port: {}", DEFAULT_PORT);
            }

            logger.info("Starting the server on {}:{}", hostname, port);

            AbstractServer server = new ProtobufServer(hostname, port, service);
            try {
                server.start();
            } catch (ServerException e) {
                logger.error("Error when trying to start the server: {}", e.getMessage());
            }

        } catch (IOException e) {
            logger.error("Error when trying to close the connection: {}", e.getMessage());
        }
    }
}
